TestApp/app/App.js

```diff
@@ -26,8 +26,20 @@ export default createBottomTabNavigator(
   }
 );
 
+let previousAppState = AppState.currentState;
+export const setPreviousAppState = (appState) => {
+  previousAppState = appState;
+  // Workaround to distinguish between notifications received in the foreground or while the app was in the background.
```

Comment:
I don't know much in RN, can you please tell me a bit more about this workaround? Also is the choice for 1000ms arbitrary?

Reply:
@jdnichollsc Can you explain what the purpose of 1000 ms timer? If I understand correctly, you're changing `previousAppState` from active to inactive when push received from background after 1000 ms, why this behavior?

Reply:
@dhei You need a period of time to know if onPushNotificationReceived is running through a notification that arrived when the application was in the background, later you need to update the status for notifications that arrive when the app is active (foreground). 1000 milliseconds is an arbitrary value, the callback should be executed in that time lapse.

Reply:
I personally tend to avoid the use of timer in such situation because the time to achieve the action we are waiting on is relative to many things like the hardware, OS version, state of the devices resources at the moment...I'd rather rely on an event fired at the right time. I know it's not always possible so in the few cases where I can't rely on an event their is also the possibility to do polling (inexpensive frequent check on the state/resource we are waiting for). But it's just my opinion and completely depends on the situation anyway.